source/index.txt

```diff
@@ -66,9 +66,14 @@ core operations. This section contains information on how to:
 API Documentation
 -----------------
 
-The `API documentation <{+api+}/apidocs/mongodb-driver-sync/>`__
-has detailed technical information about classes, methods, and
-configuration objects within the MongoDB Java driver.
+The MongoDB Java driver API documentation is split into several parts.
+Each part has detailed technical information about classes, methods, and
+configuration objects within the MongoDB Java driver:
+
+- `BSON <{+api+}/apidocs/bson/index.html>`__ (The BSON Layer)
+- `Core <{+api+}/apidocs/mongodb-driver-core/index.html>`__ (The Shared Core Classes)
+- `Java Driver <{+api+}/apidocs/mongodb-driver-sync/index.html>`__ (Modern API)
+- `Java Driver <{+api+}/apidocs/mongodb-driver-legacy/index.html>`__ (Legacy API)
```

Comment:
1) Not sure this formatting makes sense, especially using title case inside parentheses to describe each link. How about a list table instead with "Section" and "Description" as headings? Maybe you could describe them in a sentence rather than using short descriptions that sacrifice accuracy for shortness?

2) Is "layer" the correct terminology to use when describing the BSON library?

3) Just curious, but is the change that involves adding this to the "API Documentation" item in the left drawer part of this ticket? I think it would be great to include that change if it's not part of another ticket since we talked about it at the team PR review.

Reply:
1. Agreed, I'll put this in a table
2. I took that wording from the GitHub pages site, but came across "Contains the base BSON classes." in the API docs, so I'll use that instead.
3. This was done previously in [this ticket](https://jira.mongodb.org/browse/DOCSP-17236)